pds_doi_core/actions/release.py

```diff
@@ -70,13 +70,39 @@ def _parse_input(self, input):
 
         return o_doi_label
 
+    def _collect_exception_classes_and_messages(self, single_exception, io_exception_classes, io_exception_messages):
```

Comment:
Why do we need this extra piece of code ?

The error message from the exception should be accessible with str(e) where e is the exception.

Reply:
The function is necessary to collect the class names and error message of each exception.    This is different logic than exiting at first exception.

{pds-dev-el7.jpl.nasa.gov}/home/qchau/sandbox/pds-doi-service/pds_doi_core/actions 170 % grep -n -B 1 -A 1 _collect_exception_classes_and_messages *.py
release.py-72-
release.py:73:    def _collect_exception_classes_and_messages(self, single_exception, io_exception_classes, io_exception_messages):
release.py-74-        # Given a single exception, collect the exception class name and message.
--
release.py-122-                    TitleDoesNotMatchProductTypeException, SiteURNotExistException, WarningDOIException) as e:
release.py:123:                    (exception_classes, exception_messages) = self._collect_exception_classes_and_messages(e, 
release.py-124-                                                                  exception_classes, exception_messages)
{pds-dev-el7.jpl.nasa.gov}/home/qchau/sandbox/pds-doi-service/pds_doi_core/actions 171 % 

and then the exception gets raised if there is at least one.   That way, the user knows how many issues are with the DOIs so they can be corrected.

{pds-dev-el7.jpl.nasa.gov}/home/qchau/sandbox/pds-doi-service/pds_doi_core/actions 173 % grep -n -A 2 "len(exception_classes)" *.py
release.py:129:        if len(exception_classes) > 0:
release.py-130-            self._raise_warn_exceptions(exception_classes,exception_messages)
release.py-131-

Reply:
Ok, I see. Thanks